diff --git a/patched-plugins/org.eclipse.emf.edapt.common/src/org/eclipse/emf/edapt/common/ResourceUtils.java b/patched-plugins/org.eclipse.emf.edapt.common/src/org/eclipse/emf/edapt/common/ResourceUtils.java
index 16b7a51..aee2935 100644
--- a/patched-plugins/org.eclipse.emf.edapt.common/src/org/eclipse/emf/edapt/common/ResourceUtils.java
+++ b/patched-plugins/org.eclipse.emf.edapt.common/src/org/eclipse/emf/edapt/common/ResourceUtils.java
@@ -131,8 +131,10 @@
 			// Whenever we load a resourceset, we will create a session and
 			// transaction.
 			// in case we deal with a CDO Resource.
-			final Resource resource = resourceSet.getResource(modelURI, true);
-			// Resource resource = resourceSet.createResource(modelURI);
+            Resource resource = resourceSet.getResource(modelURI, false);
+            if (resource == null) {
+                resource = resourceSet.createResource(modelURI);
+            }
 			try {
 				resource.load(options);
 			} catch (final Resource.IOWrappedException e) {